README.md

```diff
@@ -65,8 +65,9 @@ Root section name can be changed:
 ```
 
 ```csharp
+var context = new ConfigurationContext { SectionName = "CustomSection" };
```

Comment:
`new ConfigurationReaderOptions()`? The "context" suffix gets a bit overloaded, especially when several of the members are just regular old options.

Reply:
I must admit that I anticipated on my next pull request which will address #206. I plan to add a new property to get access to the log level switches on that new object:

```csharp
public IReadOnlyDictionary<string, LoggingLevelSwitch> LogLevelSwitches { get; }
```

I already have a [LogLevelSwitches](https://github.com/0xced/serilog-settings-configuration/tree/LogLevelSwitches) branch on top of this `FormatProvider` branch if you want to have a look.

At that point, that `ConfigurationReaderOptions` would become «bidirectional», i.e. properties to tweak the reader behavior + property to access things (log level switches for now) created after the configuration is read. So naming it _context_ might be more appropriate I think. Maybe `ConfigurationReaderContext` context would be a better choice?

Reply:
Thanks for the reply! I think even in the "bidirectional" case, it would be more common to see callbacks on the options type, e.g.:

```csharp
new ConfigurationReaderOptions
{
    OnAddedLevelSwitch = levelSwitch => _mySwitches.Add(levelSwitch)
}
```

This would save us from needing to worry about the lifetime and thread-safety of other things attached to the options type, or having to explain that the type is bidirectional (and flag which members are "in" vs "out").

(Having named a few things `*Context` over the years, I've just about always regretted it one way or the other :-) ... for example, I'd love to go back and rename `LogContext` to the much more descriptive `LoggingScope`, or `ForContext()` to `WithTag()` or just about any other name ... context is used in so many ways that it's hard to guess its precise intention when you come across one.)

What do you think?

Reply:
Very good point! I went with `ConfigurationReaderOptions` as you suggested and force-pushed fbc5e72b06787fd7639ab9c3179c856db34ec030 on top of the `dev` branch. I'll adapt my [LogLevelSwitches](https://github.com/0xced/serilog-settings-configuration/tree/LogLevelSwitches) branch with your `OnAddedLevelSwitch` callback proposal and I'll open a new pull request to address #206 once this one is merged.

I also adapted the README to mention _since version 4.0.0_ instead of _since version 3.5.0_ as I initially did before the rebase.

Side note: how would you name the [HttpContext ](https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.http.httpcontext) class? 😂